src/pipeline/pipeline.h

```diff
@@ -61,6 +63,7 @@ namespace librealsense
 
             rs2_frame_callback_sptr _streams_callback;
             std::vector<rs2_stream> _synced_streams;
+            std::shared_ptr< librealsense::device_interface > _dev = nullptr;
```

Comment:
@OhadMeir do you think we should have a weak_ptr here?
Or we want the pipeline to have the ownership as well?

Reply:
We want the pipeline to have ownership as well.
We don't want to start working with a different device in the middle of some action. Also from user perspective options can be set without him being aware of that.

Reply:
Agreed